src/tox/tox_env/python/virtual_env/package/pyproject.py

```diff
@@ -118,8 +118,15 @@ def root(self) -> Path:
 
     @root.setter
     def root(self, value: Path) -> None:
-        self._root = value
-        self._frontend_ = None  # force recreating the frontend with new root
+        # NOTE(vytas): Recreating the frontend with a new root will orphan the
+        #   current frontend.backend_executor, if any, making tox hang upon
```

Comment:
We should instead explicitly close down that frontend.

Reply:
I've verified this approach also works with my MRE from the issue.